### ISSUE TITLE
sh <name> should always expand "file", but not "file.txt" etc.

### ISSUE THREAD

**Issue by jidanni** [scop/bash-completion#442]
```
# sh -x bin/open...
```
This should not need *.sh after its name to expand, or even need execute r-xr-xr-x permission in this case.

Back in shell script school we were never taught we needed any "." in script names.

**Comment by scop** [comment]
One could say a similar thing about pretty much all our completions. We do however choose to restrict filename completions based on common "suffixes" for the context almost everywhere, including `sh`. Users may choose otherwise if they like, see the first FAQ item in README.md.

**Comment by jidanni** [comment]
@akinomyoga indeed, that is a super duper solution. Do please reopen this bug so folks can get to work on it. Thanks.

**Comment by akinomyoga** [comment]
Sure.
